nativelib/src/main/scala/scala/scalanative/runtime/Intrinsics.scala

```diff
@@ -134,9 +139,63 @@ object Intrinsics {
   /** Intrinsified cast that reinterprets raw pointer as an long. */
   def castRawPtrToLong(rawptr: RawPtr): Long = intrinsic
 
+  /** Intrinsified cast that reinterprets raw pointer as a raw size. */
+  def castRawPtrToRawSize(rawptr: RawPtr): RawSize = intrinsic
+
   /** Intrinsified cast that reinterprets int as a raw pointer. */
   def castIntToRawPtr(int: Int): RawPtr = intrinsic
 
   /** Intrinsified cast that reinterprets long as a raw pointer. */
   def castLongToRawPtr(int: Long): RawPtr = intrinsic
+
+  /** Intrinsified cast that reinterprets raw size as an int. */
+  def castRawSizeToInt(rawSize: RawSize): Int = intrinsic
+
+  /** Intrinsified cast that reinterprets raw size as a signed long. */
+  def castRawSizeToLong(rawSize: RawSize): Long = intrinsic
+
+  /** Intrinsified cast that reinterprets raw size as an unsigned long. */
+  def castRawSizeToLongUnsigned(rawSize: RawSize): Long = intrinsic
+
+  /** Intrinsified cast that reinterprets int as a signed raw size. */
+  def castIntToRawSize(int: Int): RawSize = intrinsic
+
+  /** Intrinsified cast that reinterprets int as an unsigned raw size. */
+  def castIntToRawSizeUnsigned(int: Int): RawSize = intrinsic
+
+  /** Intrinsified cast that reinterprets long as a raw size. */
+  def castLongToRawSize(long: Long): RawSize = intrinsic
+
+  /** Intrinsic to get the size of a size in bytes, resolved to a constant in the linker */
+  def sizeOfPtr: RawSize = intrinsic
```

Comment:
Now, when have linktime conditions implemented we can replace this intrinsic with method defintion   in the same manner as is `isWindows` inside `scalanative.meta.LinktimeInfo` . This way we can save NIR tag needed to encoded `Type.SizeOfPtr`